autofill/autofillScript.ts

```diff
@@ -1,18 +1,32 @@
+// Very basic function to find closest label to input field. Used for testing below
+function labelDist(labels: Element[], input: Element): Element {
+    let minDist = Number.MAX_VALUE;
+    let minLabel = labels[0];
+    for (let i = 0; i < labels.length; i++) {
+        let dist = Math.abs(labels[i].getBoundingClientRect().top - input.getBoundingClientRect().top);
+        if (dist < minDist) {
+            minDist = dist;
+            minLabel = labels[i];
+        }
+    }
+    return minLabel;
+}
+
+
 // This file will be run on the content page, and should include all the logic for inserting text into form items.
```

Comment:
This is a really interesting way to do this. Getting the actual rendered position is really cool! I think this would be a good fallback approach.

I think the first thing we should do is simply grab the label with something like this:
`const label = document.querySelector(`label[for="${inputField.id}"]`);`

if the website has its accessibly set up correctly, or use any decent form builder these should already be set up and will probably be more robust

Reply:
i think i commented out a loop that did this with email fields. Just commented it out so it wouldnt interfere witht the testing